gcp/gcp_network.go

```diff
@@ -208,16 +208,21 @@ func (p *GCloud) buildFirewallRule(protocol string, ports []string, tag string,
 
 	if ipv6 {
 		name = fmt.Sprintf("ops-%s-rule-%s-ipv6", protocol, tag)
-		tag = tag + "-ipv6"
 	}
 
-	return &compute.Firewall{
+	f := &compute.Firewall{
 		Name:         name,
 		Description:  fmt.Sprintf("Allow traffic to %v ports %s", arrayToString(ports, "[]"), tag),
 		Allowed:      allowed,
 		TargetTags:   []string{tag},
 		SourceRanges: []string{src},
 	}
+
+	if network != "" {
+		f.Network = "global/networks/" + network
```

Comment:
According to https://pkg.go.dev/google.golang.org/api/compute/v1#Firewall:
> If you choose to specify this field, you can specify the network as a full or partial URL. For example, the following are all valid URLs:
> - https://www.googleapis.com/compute/v1/projects/myproject/global/networks/my-network
> - projects/myproject/global/networks/my-network
> - global/networks/default

So it appears that when specifying a non-default network you have to use either the full URL or the `projects/myproject/` prefix. In any case, apparently we need to retrieve the ProjectID from the CloudConfig.